modules/packages/UnitTest.chpl

```diff
@@ -399,21 +399,8 @@ module UnitTest {
           if all(seq1 == seq2) then return;
         }
         tmpString = seq_type_name+"s differ: ";
-        if seq_type_name == "Array" {
-          tmpString += "'[";
-          for i in 1..seq1.size {
-            if i != seq1.size then tmpString+= seq1[i]:string+", ";
-            else tmpString += seq1[i]:string+"]' != '[";
-          }
-          for i in 1..seq2.size {
-            if i != seq2.size then tmpString+= seq2[i]:string+", ";
-            else tmpString += seq2[i]:string+"]'";
-          }
-        }
-        else {
-          tmpString += "'"+stringify(seq1)+"' != '"+stringify(seq2)+"'" ;
-        }
-        for i in 1..min(len1,len2) {
+        tmpString += "'"+stringify(seq1)+"' != '"+stringify(seq2)+"'" ;
+        for i in seq1.indices.low..min(len1,len2) {
```

Comment:
If these sequences become 0-based, this will dereference `seq1[0]` through `seq1[len1]` (assuming seq1 is the shorter), which is one too many.   `seq1.indices.low..#min(len1,len2)`?

The comment at the top of the function says it can be used for ranges, but ranges don't have a `.indices`.

Reply:
That's right. There will still be some changes required in the 0-based branches, but this PR aims to reduce the changes needed significantly.

> The comment at the top of the function says it can be used for ranges, but ranges don't have a .indices.

That comment is outdated. Ranges use `__baseAssertEqual()`.